### PR TITLE
11799 edit profile

### PR DESCRIPTION
**Pull request by tuananhp-1844**
https://edu-redmine.sun-asterisk.vn/issues/11799

### CODE REVIEW

File: app/Http/Traits/Upload.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<?php
+namespace App\Http\Traits;
+
```
Comment (framgiaciv3) [inline app/Http/Traits/Upload.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Namespaces.UseDeclaration.SpaceAfterLastUse</br>**Severity**: error</br>**File**: app/Http/Traits/Upload.php</br>**Line**: 5</br>There must be one blank line after the last USE statement; 0 found;

File: app/Http/Traits/Upload.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+use Image;
+use Storage;
```
Comment (framgiaciv3) [inline app/Http/Traits/Upload.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.SuperfluousWhitespace.EndLine</br>**Severity**: error</br>**File**: app/Http/Traits/Upload.php</br>**Line**: 7</br>Whitespace found at end of line

File: app/Http/Traits/Upload.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+/**
+ * 
+ */
+trait Upload
+{
+    public function upload($file, $path)
```
Comment (framgiaciv3) [inline app/Http/Traits/Upload.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Http/Traits/Upload.php</br>**Line**: 13</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Http/Traits/Upload.php [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        // try {
+        //     if (!is_dir($path . '_custom/')) {
+        //         mkdir($path . '_custom/');
+        //     }
+        //     $img = Image::make($file->getRealPath())
+        //         ->resize(200, 200);
+        //     Storage::disk('public')->putFileAs($path, $img, $filename);
+        //     $data["image_custom"] = $path . '_custom/' . $filename;
+        // } catch (\Exception $e) {
+        // }
```
Comment (framgiaciv3) [inline app/Http/Traits/Upload.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Traits/Upload.php</br>**Line**: 30</br>String "image" does not require double quotes; use single quotes instead

File: app/Http/Traits/Upload.php [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        Storage::disk('public')->putFileAs($path, $file, $filename);
+        $data["image"] = 'storage/' . $path . '/' . $filename;
+
```
Comment (framgiaciv3) [inline app/Http/Traits/Upload.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Http/Traits/Upload.php</br>**Line**: 33</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Http/Traits/Upload.php (hunk 1 above)
Comment (framgiaciv3) [inline app/Http/Traits/Upload.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Namespaces.UseDeclaration.SpaceAfterLastUse</br>**Severity**: error</br>**File**: app/Http/Traits/Upload.php</br>**Line**: 5</br>There must be one blank line after the last USE statement; 0 found;

File: app/Http/Traits/Upload.php (hunk 2 above)
Comment (framgiaciv3) [inline app/Http/Traits/Upload.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.SuperfluousWhitespace.EndLine</br>**Severity**: error</br>**File**: app/Http/Traits/Upload.php</br>**Line**: 7</br>Whitespace found at end of line

File: app/Http/Traits/Upload.php (hunk 3 above)
Comment (framgiaciv3) [inline app/Http/Traits/Upload.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Http/Traits/Upload.php</br>**Line**: 13</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Http/Traits/Upload.php (hunk 4 above)
Comment (framgiaciv3) [inline app/Http/Traits/Upload.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Traits/Upload.php</br>**Line**: 30</br>String "image" does not require double quotes; use single quotes instead

File: app/Http/Traits/Upload.php (hunk 5 above)
Comment (framgiaciv3) [inline app/Http/Traits/Upload.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Http/Traits/Upload.php</br>**Line**: 33</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Http/Traits/Upload.php [hunk 6]
```diff
@@ -0,0 +1,35 @@
+<?php
+namespace App\Http\Traits;
+
```
Comment (framgiaciv3) [inline app/Http/Traits/Upload.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Namespaces.UseDeclaration.SpaceAfterLastUse</br>**Severity**: error</br>**File**: app/Http/Traits/Upload.php</br>**Line**: 5</br>There must be one blank line after the last USE statement; 0 found;

File: app/Http/Traits/Upload.php [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+use Image;
+use Storage;
```
Comment (framgiaciv3) [inline app/Http/Traits/Upload.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.SuperfluousWhitespace.EndLine</br>**Severity**: error</br>**File**: app/Http/Traits/Upload.php</br>**Line**: 7</br>Whitespace found at end of line

File: app/Http/Traits/Upload.php [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+        //         ->resize(200, 200);
+        //     Storage::disk('public')->putFileAs($path, $img, $filename);
+        //     $data["image_custom"] = $path . '_custom/' . $filename;
+        // } catch (\Exception $e) {
+        // }
+        Storage::disk('public')->putFileAs($path, $file, $filename);
+        $data['image'] = 'storage/' . $path . '/' . $filename;
+
+        return $data;
+    }
```
Comment (framgiaciv3) [inline app/Http/Traits/Upload.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Classes.ClassDeclaration.CloseBraceAfterBody</br>**Severity**: error</br>**File**: app/Http/Traits/Upload.php</br>**Line**: 35</br>The closing brace for the trait must go on the next line after the body

File: app/Http/Traits/Upload.php [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
```
Comment (framgiaciv3) [inline app/Http/Traits/Upload.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Classes.ClassDeclaration.CloseBraceAfterBody</br>**Severity**: error</br>**File**: app/Http/Traits/Upload.php</br>**Line**: 33</br>The closing brace for the trait must go on the next line after the body

**Comment by tuananhp-1844** [comment]
ready

**Pull request merged**